site/src/components/hooks/useShortnameTextField.ts

```diff
@@ -0,0 +1,77 @@
+import { debounce } from "lodash";
+import { useState, useEffect, useMemo } from "react";
+import { apiClient } from "../../lib/apiClient";
+import {
+  ApiIsShortnameTakenQueryParams,
+  ApiIsShortnameTakenResponse,
+} from "../../pages/api/isShortnameTaken.api";
+
+const parseShortnameInput = (input: string) =>
+  input.replaceAll(/[^a-zA-Z0-9-_]/g, "").toLowerCase();
+
+type UseShortnameTextFieldProps = {
+  touched: boolean;
+};
+
+export const useShortnameTextField = ({
+  touched,
+}: UseShortnameTextFieldProps) => {
+  const [shortname, setShortname] = useState<string>("");
+  const [isTaken, setIsTaken] = useState<boolean>();
+
+  const isEmpty = !shortname;
+
+  const isTooShort = shortname.length < 4;
+
+  const isTooLong = shortname.length > 24;
+
+  const shortnameInputIsValid = !(isEmpty || isTooShort || isTooLong);
+
+  const debouncedCheckIfShortnameIsValid = useMemo(
+    () =>
+      debounce(async (params: { shortname: string }) => {
+        const { data: isShortnameTaken, error } = await apiClient.get<
+          ApiIsShortnameTakenResponse,
+          ApiIsShortnameTakenQueryParams
+        >("isShortnameTaken", params);
+
+        if (error) {
+          throw error;
+        } else if (isShortnameTaken !== undefined) {
```

Comment:
Why compare to `undefined` rather than for truthiness? 

If `null` or `false` ever end up as a value here for `isShortnameTaken`, it'll show as taken. Appreciate that's not how it is but might trip us up later.

Reply:
I'm not understanding, `isShortnameTaken` here is the response from the server (either `true`, `false` or `undefined` if an error occurred). If we were to use truthiness, `isTaken` would never be set to `false`, and we couldn't indicate to the user their shortname is taken.

Not sure if I've misunderstood something though

Reply:
Ahh sorry, I misunderstood `setIsTaken`.